### PR TITLE
Use JSON schema for the config

### LINKED ISSUE #870: Upgrade `electron-store` and use JSON Schema
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/caprine/issues/870)

TypeScript will help us at compile-time, but would be nice to have runtime validation too.

https://github.com/sindresorhus/electron-store#schema

The latest version of `electron-store` also ships with its own TS definition, so we can drop the `@types/electron-store` dependency then.

> There is a $60.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/caprine/issues/870).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/caprine/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by whitecrownclown**
Fixes #870

### CODE REVIEW

File: source/config.ts [hunk 1]
```diff
@@ -1,50 +1,146 @@
 import Store = require('electron-store');
 import {is} from 'electron-util';
-import {EmojiStyle} from './emoji';
+import {JSONSchema} from 'json-schema-typed';
```
Comment (sindresorhus) [inline source/config.ts]: I think we should re-export this type from `conf` and then `electron-store` instead.

**Comment by sindresorhus** [comment]
Travis is failing

File: source/config.ts (hunk 1 above)
Comment (whitecrownclown) [reply source/config.ts]: You mean adding and exporting `json-schema-typed` to `conf` and then get it into `electron-store` through it?

Comment (sindresorhus) [reply source/config.ts]: Yes, it's weird having to import an extra package just to get the type. `conf` and `electron-store` should provide the necessary types.

Comment (sindresorhus) [reply source/config.ts]: https://github.com/sindresorhus/conf/issues/84

**Pull request merged**